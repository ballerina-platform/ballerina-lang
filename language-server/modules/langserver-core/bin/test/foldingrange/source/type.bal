type Student record {
    string Name;
    int Grade;
    map<any> Marks;
};

# Define a closed record type named `Address`. The `{|` and `|}` delimiters indicate that this record type
# allows mapping values, which contain only the described fields.
type Address record {|
    string city;
    string country;
|};

# Description
#
# + age - age Parameter Description
# + firstName - firstName Parameter Description
# + lastName - lastName Parameter Description
type Person object {
    public int age;
    public string firstName;
    public string lastName;

    // Returns full name
    function getFullName() returns string;

    # Description
    #
    # + condition - condition Parameter Description
    # + a - a Parameter Description
    function checkAndModifyAge(int condition, int a);
};
