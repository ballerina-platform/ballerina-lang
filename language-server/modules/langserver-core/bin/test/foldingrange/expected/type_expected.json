{
  "result":[
    {
      "startLine":0,
      "endLine":3,
      "kind":"region"
    },
    {
      "startLine":6,
      "endLine":7,
      "kind":"comment"
    },
    {
      "startLine":8,
      "endLine":10,
      "kind":"region"
    },
    {
      "startLine":13,
      "endLine":17,
      "kind":"comment"
    },
    {
      "startLine":18,
      "endLine":30,
      "kind":"region"
    },
    {
      "startLine":24,
      "endLine":24,
      "kind":"region"
    },
    {
      "startLine":30,
      "endLine":30,
      "kind":"region"
    },
    {
      "startLine":26,
      "endLine":29,
      "kind":"comment"
    }
  ]
}
